src/overlays/actors/ovl_En_Dh/z_en_dh.c

```diff
@@ -21,47 +54,473 @@ const ActorInit En_Dh_InitVars = {
+            case 0:
+                this->actor.flags |= 1;
+                this->actor.shape.rot.y = this->actor.yawTowardsLink;
+                this->actor.flags &= ~0x80;
+                this->actionState++;
+                this->drawDirtWave++;
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEADHAND_HIDE);
+            case 1:
+                this->dirtWavePhase += 0x3A7;
+                Math_SmoothScaleMaxMinF(&this->dirtWaveSpread, 300.0f, 1.0f, 5.0f, 0.0f);
+                this->dirtWaveHeight = Math_Sins(this->dirtWavePhase) * 55.0f;
+                this->dirtWaveAlpha = (s16)(Math_Sins(this->dirtWavePhase) * 255.0f);
+                EnDh_SpawnDebris(globalCtx, this, &this->actor.posRot.pos, this->dirtWaveSpread, 4, 2.05f, 1.2f);
+                if (this->actor.shape.unk_08 == 0.0f) {
+                    this->drawDirtWave = false;
+                    this->actionState++;
+                } else if (this->dirtWavePhase > 0x12C0) {
+                    this->actor.shape.unk_08 += 500.0f;
+                }
+                break;
+            case 2:
+                EnDh_SetupWalk(this);
+                break;
+        }
+        Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->actor.yawTowardsLink, 1, 0x7D0, 0);
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+        if (this->actor.params != ENDH_START_ATTACK_BOMB) {
+            func_8008EEAC(globalCtx, &this->actor);
+        }
+    }
+}
+
+void EnDh_SetupWalk(EnDh* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06003A8C, 1.0f, 0.0f,
+                         SkelAnime_GetFrameCount(&D_06003A8C.genericHeader) - 3.0f, 0, -6.0f);
+    this->curAction = ACTION_WALK;
+    this->timer = 300;
+    this->actor.speedXZ = 1.0f;
+    EnDh_SetupAction(this, EnDh_Walk);
+}
+
+void EnDh_Walk(EnDh* this, GlobalContext* globalCtx) {
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->actor.yawTowardsLink, 1, 0xFA, 0);
+    this->actor.posRot.rot.y = this->actor.shape.rot.y;
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if ((((s32)this->skelAnime.animCurrentFrame) % 8) == 0) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEADHAND_WALK);
+    }
+    if ((globalCtx->gameplayFrames & 0x5F) == 0) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEADHAND_LAUGH);
+    }
+    if (this->actor.xzDistFromLink <= 100.0f) {
+        this->actor.speedXZ = 0.0f;
+        if (func_8002E084(&this->actor, 60 * 0x10000 / 360)) {
+            EnDh_SetupAttack(this);
+        }
+    } else if (--this->timer == 0) {
+        EnDh_SetupBurrow(this);
+    }
+}
+
+void EnDh_SetupRetreat(EnDh* this, GlobalContext* globalCtx) {
+    SkelAnime_ChangeAnimTransitionRepeat(&this->skelAnime, &D_06005880, -4.0f);
+    this->curAction = ACTION_RETREAT;
+    this->timer = 70;
+    this->actor.speedXZ = 1.0f;
+    EnDh_SetupAction(this, EnDh_Retreat);
+}
+
+void EnDh_Retreat(EnDh* this, GlobalContext* globalCtx) {
+    if (--this->timer == 0) {
```

Comment:
pre decrement required to match? imo a decrement separate from the comparison is nicer

Reply:
not required. fixed. also cleaned up a few other things and formatted.